src/Parser.js

```diff
@@ -74,27 +76,27 @@ export function parse (value, root, rule, rules, rulesets, points, declarations)
 					case 59: characters += ';'
 					// { rule/at-rule
 					default:
-						append(reference = ruleset(characters, root, index, offset, rules, points, type, props = [], children = [], length), rulesets)
+						append(reference = ruleset(characters, root, index, offset, rules, points, type, props = [], children = [], (pseudo = pseudo || inpseudo)), rulesets)
```

Comment:
Can you try this diff: https://gist.github.com/thysultan/186b10150fce56d40f7cac2b165780c8#file-parser-js (updated)

Reply:
it doesn't work for [the orphaned `:read-write` case](https://github.com/thysultan/stylis.js/blob/c4159ecb3a88af498d9a2128e0306f81756c1da1/test/Middleware.js#L77-L80). The problem here is that `:` comes first and thus `length` (used as `pseudo`) is not initialized to a positive integer - it's always 0 for a case like this - and that makes it hard to reuse `length` as `pseudo`.

I've also checked size of both approaches - ur has at the moment 9 bytes less than mine

Reply:
can't we use the property variable?

Reply:
for example:

Parser.js#72

```diff
- append(property > 32 ? declaration(characters + ';', rule, length - 1) : declaration(replace(characters, ' ', '') + ';', rule, length - 2), declarations)
+ append(property > 32 ? declaration(characters + ';', rule, length - 1) : declaration(replace(characters, ' ', '') + ';', rule, length - 2), declarations)
```

Parser.js#98

```diff
- length = strlen(characters), property = previous
+ length = 1 + strlen(characters), property = previous
```

Reply:
I've applied your fix - although overall I think mine was more clear (if it has worked for all cases) and it had this cool effect of `.length` being 0 or 1 for rules. Stylis is all about squeezing the most we can and your version is better by 8 bytes, so I'm fine with that 😉

Reply:
LGTM